source/vibe/core/drivers/libasync.d

```
@@ -105,8 +105,13 @@ final class LibasyncDriver : EventDriver {
 
 	}
 
-	static @property bool isControlThread() {
-		return Thread.getThis().isDaemon && Thread.getThis().name == "CmdProcessor" ;
+	static @property bool isControlThread() nothrow {
+		bool is_ctrl_thr;
+
+		try is_ctrl_thr = Thread.getThis().isDaemon && Thread.getThis().name == "CmdProcessor";
+		catch {}
```

`scope (failure) assert(false);` instead of just ignoring the error.